File: INTERIF/organograma.cs
using System;
using System.Collections.Generic;

class Program
{
    static void Main()
    {
        string nad = Console.ReadLine();

        Dictionary<string, List<string>> graph = new Dictionary<string, List<string>>();

        while (true)
        {
            string[] edge = Console.ReadLine().Split(' ');
            if (edge[0] == "fim" && edge[1] == "entrada")
            {
                break;
            }
            if (!graph.ContainsKey(edge[1]))
            {
                graph[edge[1]] = new List<string>();
            }
            graph[edge[1]].Add(edge[0]);
        }

        string start = Console.ReadLine();


        List<string> fila = new List<string> { start };
        List<string> processados = new List<string>();


        while (fila.Count > 0)
        {
            string current = fila[0];
            if (graph.ContainsKey(current))
            {
                List<string> paordenar = new List<string>();
                foreach (string i in graph[current])
                {
                    if (!processados.Contains(i))
                    {
                        paordenar.Add(i);
                    }
                }

                paordenar.Sort();
                paordenar.Reverse();


                foreach (string i in paordenar)
                {
                    fila.Insert(1, i);
                }
            }

            Console.WriteLine(current);

            processados.Add(current);
            fila.RemoveAt(0);
        }
    }
}

File: INTERIF/organograma.py

nad=input()
graph={}
while True:
    edge=input().split(' ')
    if edge == ['fim', 'entrada']:
        break
    if not edge[1] in graph:
        graph[edge[1]]=[]
    graph[edge[1]].append(edge[0])
start=input()
fila=[start]
processados=[]
while fila :   
    current=fila[0]
    if current in graph:
        paordenar=[]
        for i in graph[current]:
            if not i in processados:
                paordenar.append(i)
        paordenar.sort(reverse=True)
        for i in paordenar:
            fila.insert(1, i)
    print(current)
    processados.append(current)
    fila.pop(0)

